include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
+		const Gaffer::StringPlug *colorSpaceMetadataNamePlug() const;
+
 		virtual void affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const;
 
 		static size_t supportedExtensions( std::vector<std::string> &extensions );
 
+		typedef boost::function<const std::string ( const IECore::CompoundData * )> DefaultColorSpaceFunction;	
+		static void registerDefaultColorSpace( DefaultColorSpaceFunction f );
```

Comment:
ideally we would have the exact same function used in both case, but as you can see from my default implementation that doesnt quite work ( dataType name not consistent between read and write ).Beside I think Gaffer should offer the flexibility to have very different config for reader and writer.Since we also provide an example where the same config is registered for both node, we support also the simpler approach.